master/internal/api_master.go

```diff
@@ -30,25 +30,25 @@ func (a *apiServer) GetMaster(
 	_ context.Context, _ *apiv1.GetMasterRequest,
 ) (*apiv1.GetMasterResponse, error) {
 	product := apiv1.GetMasterResponse_PRODUCT_UNSPECIFIED
-	if a.m.config.InternalConfig.ExternalSessions.Enabled() {
+	if a.m.config.Load().InternalConfig.ExternalSessions.Enabled() {
```

Comment:
for this endpoint, let's load the config once upfront and reuse that loaded copy so the read is atomic.